packages/store/src/store.ts

```diff
@@ -116,12 +106,7 @@ export class Store {
       selectorFn
     );
 
-    try {
-      assertInInjectionContext(() => {});
-      return toSignal(observable, { initialValue });
-    } catch {
-      return toSignal(observable, { initialValue, injector: this._injector });
-    }
+    return toSignal(observable, { initialValue });
```

Comment:
Oh nice, so this is able to determine the correct injection context at run time🎉.